pandas/tests/series/methods/test_astype.py

```diff
@@ -350,6 +350,39 @@ def test_astype_bytes(self):
         assert result.dtypes == np.dtype("S3")
 
 
+class TestAstypeString:
+    @pytest.mark.parametrize(
+        "data, dtype",
+        [
+            (["A", NA], "category"),
+            (["2020-10-10", "2020-10-10"], "datetime64[ns]"),
+            (["2020-10-10", "2020-10-10", NaT], "datetime64[ns]"),
+            (
+                ["2012-01-01 00:00:00-05:00", NaT],
+                "datetime64[ns, US/Eastern]",
+            ),
+            ([1, None], "UInt16"),
+            (["1/1/2021", "2/1/2021"], "period[M]"),
+            (["1/1/2021", "2/1/2021", NaT], "period[M]"),
+            (["1 Day", "59 Days", NaT], "timedelta64[ns]"),
+            # currently no way to parse BooleanArray, IntervalArray from a
```

Comment:
BooleanArray can be parsed from string (see _from_sequence_of_strings, the general method)

Reply:
I had an implementation that used `_from_sequence_of_strings` instead of `_from_sequence` to in StringArray.astype(). That required bigger code changes. I'd like to merge this Regression PR and then implement to implement `_from_sequence_of_strings` as part of #40566